mcs/class/System.Windows.Forms/System.Windows.Forms/StatusBar.cs

```diff
@@ -220,15 +220,22 @@ protected override void CreateHandle ()
 		}
 
 		protected override void Dispose (bool disposing) {
-			if (!IsDisposed)
+			if (!IsDisposed && disposing)
 			{
-				if (disposing)
-				{
-					if (tooltip_timer != null)
-						tooltip_timer.Dispose();
+				if (tooltip_timer != null)
+					tooltip_timer.Dispose();
 
-					if (tooltip_window != null)
-						tooltip_window.Dispose();
+				if (tooltip_window != null)
+					tooltip_window.Dispose();
+
+				if (panels != null) {
+					var copiedPanels = new StatusBarPanel [panels.Count];
```

Comment:
Why is this copy needed?

Reply:
Good question. I copied that part from the referencesource implementation. I guess it's theoretically possible that calling `panel.Dispose()` modifies `panels`.